src/main/java/com/example/kadai9/NameMapper.java

```diff
@@ -12,5 +12,5 @@ public interface NameMapper {
     List<Name> findAll();
 
     @Select("SELECT * FROM names WHERE id = #{id}")
-    Optional<Name> findById(int id);
+    Name findById(int id);
```

Comment:
Optionalについて調べるということでしたのでぜひ調べた上で下記のように変更するとよいです。
```suggestion
    Optional<Name> findById(int id);
```

そしてMapperのfindByIdを呼び出すServiceImplの実装では
```java
public Name findById(int id) {
  Optional<Name> name = nameMapper.findById(id);
  if (name.isPresent) {
    return name.get();
  } else {
    throw new 名前が見つからなかったとき用Exception("name not found for id: " + id);
  }
}
```
のように実装するとよいです。
Optionalは主にメソッドの戻り値が1つの値もしくは空になりうることを明示するために使います。
ID検索では1件のレコードがHITするか、存在しないIDなら0件のレコードが返されるかの2つの結果が予想できます。
ですのでここではOptionalを使うのが妥当だと考えています。

なお、先の実装例ですがorElseThrowというOptionalのメソッドを利用することでより短く処理がかけますので良ければ試してみてください。